wallet/types/src/keys.rs

```diff
@@ -130,6 +132,60 @@ pub struct RootKeys {
     pub root_vrf_key: ExtendedVRFPrivateKey,
 }
 
+/// Just an empty struct used as key for the DB table
+/// It only represents a single value as there can be only one root key
+#[derive(PartialEq, Eq, PartialOrd, Ord, Encode, Decode)]
+pub struct SeedPhraseConstant;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SeedPhrase {
+    pub mnemonic: zeroize::Zeroizing<bip39::Mnemonic>,
+}
+
+impl Encode for SeedPhrase {
+    fn encode(&self) -> Vec<u8> {
+        let mut buf = BufWriter::new(Vec::new());
+        self.encode_to(&mut buf);
+        buf.into_inner().expect("Flushing should never fail")
+    }
+
+    fn size_hint(&self) -> usize {
+        // Preallocate enough space for the longest possible word list
+        33
+    }
+
+    fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+        f(self.encode().as_slice())
+    }
+
+    fn encoded_size(&self) -> usize {
+        self.encode().len()
+    }
+
+    fn encode_to<T: serialization::Output + ?Sized>(&self, dest: &mut T) {
+        let entropy = self.mnemonic.to_entropy();
```

Comment:
Are you sure using `to_entropy` is the correct way to encode thise? Why not use the serialized version of the struct. Looking briefly in it made it seem that 64-bytes do contain everything.

My concern is that the entropy you're serializing is not necessarily backward/forward compatible with changes in the library and is not part of the standard. We should serialize things that are specifically part of the standard to guarantee that things will never break.

Reply:
I am now encoding the seed phrase words.